src/encoding.md

```diff
@@ -491,6 +492,84 @@ alisha,colombo,xyz";
 # quick_main!(run);
 ```
 
+[ex-csv-read]: #ex-csv-read
+<a name="ex-csv-read"></a>
+## Read CSV records
+
+[![csv-badge]][csv] [![cat-encoding-badge]][cat-encoding]
+
+Reads standard CSV records into [`csv::StringRecord`] (valid UTF-8 bytes).
+
+```rust
+extern crate csv;
+# #[macro_use]
+# extern crate error_chain;
+#
+# error_chain! {
+#     foreign_links {
+#         Reader(csv::Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let csv = "year,make,model,description
+1948,Porsche,356,Luxury sports car
+1967,Ford,mustang fastback 1967,American car";
+
+    let mut reader = csv::Reader::from_reader(csv.as_bytes());
+    for record in reader.records() {
+        println!("{:?}", record?);
+    }
+
+    Ok(())
+}
+#
+# quick_main!(run);
+```
+
+This is like the previous example, however Serde is used to
+deserialize data into strongly type structures. See the
+[`csv::Reader::deserialize`] method.
+
+```rust
+extern crate csv;
+# #[macro_use]
+# extern crate error_chain;
+#[macro_use]
+extern crate serde_derive;
+
+# error_chain! {
+#     foreign_links {
+#         Reader(csv::Error);
+#     }
+# }
+#
+fn run() -> Result<()> {
+    let csv = "year,make,model,description
+1948,Porsche,356,Luxury sports car
+1967,Ford,mustang fastback 1967,American car";
+
+    let mut reader = csv::Reader::from_reader(csv.as_bytes());
+
+    for record in reader.deserialize() {
+        let record: Record = record?;
+        println!("{:?}", record);
+    }
+
+    Ok(())
+}
+
+#[derive(Debug, Deserialize)]
+struct Record {
```

Comment:
I would move the struct definition to the beginning of the snippet. this will improve the reading flow somewhat